contracts/pcv/utils/FeiSkimmer.sol

```diff
@@ -3,69 +3,106 @@ pragma solidity ^0.8.4;
-    IPCVDeposit public source;
+    EnumerableSet.AddressSet private sources;
 
     /// @notice the threshold of FEI above which to skim
-    uint256 public threshold;
+    mapping(address => uint256) public thresholds;
 
     /// @notice FEI Skimmer
     /// @param _core Fei Core for reference
-    /// @param _source the target to skim from
-    /// @param _threshold the threshold of FEI to be maintained by source
+    /// @param _sources the targets to skim from
+    /// @param _thresholds the thresholds of FEI to be maintained by sources
     constructor(
         address _core,
-        IPCVDeposit _source,
-        uint256 _threshold
+        address[] memory _sources,
+        uint256[] memory _thresholds
     ) 
         CoreRef(_core)
     {
-        source = _source;
-        threshold = _threshold;
-        emit ThresholdUpdate(threshold);
+        require(_sources.length == _thresholds.length, "array length mismatch");
+        require(_sources.length > 0, "must provide at least once source");
+        
+        for(uint i=0; i<_sources.length; i++) {
+            sources.add(_sources[i]);
+            thresholds[_sources[i]] = _thresholds[i];
+            emit SourceAdded(_sources[i], _thresholds[i]);
+        }
+    }
+
+    /// @notice convenience method to view sources & thresholds
+    function getSources() external view returns (address[] memory sourceAddresses, uint256[] memory sourceThresholds) {
+        sourceAddresses = new address[](sources.length());
+        sourceThresholds = new uint256[](sources.length());
+
+        for (uint i=0; i<sources.length(); i++) {
+            sourceAddresses[i] = sources.at(i);
+            sourceThresholds[i] = thresholds[sources.at(i)];
+        }
+
+        return (sourceAddresses, sourceThresholds);
     }
 
     /// @return true if FEI balance of source exceeds threshold
-    function skimEligible() external view returns (bool) {
-        return fei().balanceOf(address(source)) > threshold;
+    function skimEligible(address _source) public view returns (bool) {
+        require(sources.contains(_source), "source not valid");
+        return fei().balanceOf(_source) > thresholds[_source];
     }
 
     /// @notice skim FEI above the threshold from the source. Pausable. Requires skimEligible()
-    function skim()
-        external
+    function skim(address _source)
+        public
         whenNotPaused
     {
-        IFei _fei = fei();
-        uint256 feiTotal = _fei.balanceOf(address(source));
-
-        require(feiTotal > threshold, "under threshold");
-        
-        uint256 burnAmount = feiTotal - threshold;
-        source.withdrawERC20(address(_fei), address(this), burnAmount);
+        if(skimEligible(_source)) {
+            IFei fei = fei();
+            uint256 burnAmount = fei.balanceOf(_source) - thresholds[_source];
+            IPCVDeposit(_source).withdrawERC20(address(fei), address(this), burnAmount);
+            fei.burn(burnAmount);
+        }
```

Comment:
not super gas efficient to check skimEligible for everything. I recommend following a similar pattern to the pre-existing skimmer

Reply:
also you should separate out a burn function and call burn at the end, in the skimAll case